optuna/integration/botorch.py

```diff
@@ -670,9 +670,20 @@ def sample_relative(
             # `manual_seed` makes the default candidates functions reproducible.
             # `SobolQMCNormalSampler`'s constructor has a `seed` argument, but its behavior is
             # deterministic when the BoTorch's seed is fixed.
-            candidates = self._candidates_func(
-                completed_params, completed_values, con, bounds, running_params
-            )
+            try:
+                candidates = self._candidates_func(
+                    completed_params, completed_values, con, bounds, running_params
+                )
+            except TypeError:
+                if self._consider_running_trials:
+                    warnings.warn(
+                        f"The provided candidates_func {self._candidates_func.__name__} must take "
+                        "an argument pending_x to support option consider_running_trials=True. "
+                        "consider_running_trials option will be ignored."
+                    )
+                candidates = self._candidates_func(
+                    completed_params, completed_values, con, bounds
+                )  # type: ignore
```

Comment:
Getting the mypy check right here would require even more lines of code, so I'd use `# type: ignore`.